services/algolia-sync/src/components/aws-sqs/sqs.js

```diff
@@ -0,0 +1,35 @@
+require('aws-sdk/clients/sqs');
+const AWS = require('aws-sdk/global');
+const algolia = require('../algolia/message-sync');
+const cfg = require('./config');
+const { AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, AWS_REGION } = require('../../env');
+
+// Create the SQS service object
+const sqs = new AWS.SQS(
+  {
+    region: AWS_REGION,
+    apiVersion: '2012-11-05',
+    accessKeyId: AWS_ACCESS_KEY_ID,
+    secretAccessKey: AWS_SECRET_ACCESS_KEY,
+  },
+);
+// Get's a message off the the queue
+const receiveMessage = async () => new Promise(async (resolve, reject) => {
```

Comment:
You can use the `receiveMessage` and `deleteMessage` as promises using the `.promise()` call. As such, there's no need to wrap this. Instead approach it like this:

```js
const receiveMessage = async () => {
  const data = await sqs.receiveMessage(cfg.receive()).promise();
  if (data.Messages) {
    await algolia.upsertToIndex(JSON.parse(data.Messages[0].Body));
    await sqs.deleteMessage(cfg.delete(data)).promise();
  }
};
```

Reply:
Thanks for the feedback, I felt like it was messy. Just to make sure I understand this. The .promise() will handle the cache part for me and throw the error?

Reply:
You can either use a callback or use the `.promise()` chain to return a promise. The promise can then be awaited and tried/caught for error handling. Otherwise, you will have to handle errors within your callback.

Not sure what you mean by cache?

Reply:
Haha need more coffee I guess. I was trying to say catch.